CONTRIBUTING.md

```diff
@@ -50,6 +50,47 @@ paper and understand how it fits into the TensorFlow community. This
 process can take longer than typical commit reviews so please bare with
 us**
 
+## Tools needed for developement
+
+#### Linux
+
+* Docker (code formatting / testing)
+* Nvidia-docker (for GPU testing, optional)
+* Bazel installed locally (to build custom ops locally, optional)
+* NVCC/Cuda installed locally (to build custom ops with gpu locally, optional)
+
+#### MacOS
+
+* Docker (code formatting / testing)
+* Bazel installed locally (to build custom ops locally, optional)
+
+#### Windows
+
+For Windows, you have two options:
+
+##### WSL 2
+
+WSL 2 is a very light virtual machine running with hyper-V. When running in 
+WSL 2, you're in a full linux environment, with a real linux kernel. 
+WSL 2 networking is shared with Windows and your Windows files can be found under 
+`/mnt/c`. When working with WSL 2, you can just follow the linux guides and tutorials
+and everything will work as in linux, including 
+Docker (that means you install docker with apt-get), git, ssh...
+See [the WSL 2 install guide](https://docs.microsoft.com/en-us/windows/wsl/wsl2-install).
+
+##### Powershell in Windows
+
+This is if you want to stay in Windows world. In this case, you need:
+
+* Git with git bash install on the PATH (meaning that you can run the `sh` command from Powershell).
+* Docker desktop with Linux containers (code format, testing on linux, etc...)
+* A local Python installation
+* Bazel (if you want to compile custom ops on Windows, optional)
+* [Visual Studio build tools 2017](https://devblogs.microsoft.com/cppblog/visual-studio-build-tools-now-include-the-vs2017-and-vs2015-msvc-toolsets/)
```

Comment:
nit: I believe we're using MSVC2019 which aligns with TF-core:
https://www.tensorflow.org/install/source_windows#install_visual_c_build_tools_2019